cmd/query/app/util.go

```diff
@@ -14,7 +14,31 @@
 
 package app
 
-import "github.com/jaegertracing/jaeger/storage/spanstore"
+import (
+	"github.com/jaegertracing/jaeger/cmd/query/app/querysvc"
+	"github.com/jaegertracing/jaeger/model/adjuster"
+	"github.com/jaegertracing/jaeger/storage"
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+	"go.uber.org/zap"
+)
+
+// BuildQueryServiceOptions creates a QueryServiceOptions struct with appropriate adjusters and archive config
+func BuildQueryServiceOptions(storageFactory storage.Factory, queryOptions *QueryOptions, logger *zap.Logger) *querysvc.QueryServiceOptions {
```

Comment:
unless there are circular package import issues, I would define this function on the `*QueryOptions` receiver.